pageserver/src/tenant.rs

```diff
@@ -133,7 +132,7 @@ pub struct Tenant {
     tenant_conf: Arc<RwLock<TenantConfOpt>>,
 
     tenant_id: TenantId,
-    timelines: Mutex<HashMap<TimelineId, Arc<Timeline>>>,
+    timelines: parking_lot::Mutex<HashMap<TimelineId, Arc<Timeline>>>,
```

Comment:
This worries me because we need the poison ability, in `delete_timeline` for example. Though, that can and should be fixed in an other PR.

I don't see why not make these into async mutexes? They'd miss poisoning but I didn't understand why Send able were needed.

Reply:
They need to be `Send` because we hold them across await points inside of task_mgr tasks. task_mgr tasks need to be send. See the `Implementation` section of this PR:

> 
> This PR conversts Tenant::timelines and Timeline::layers from std::sync to parking_lot primitives.
> We use parking lot with the send_guard feature enabled.
> That makes the Mutex and RwLock guard objects Send.
> Having them Send allows us to hold them across .await points.